### PR TITLE
[4.0][WIP] Field tag

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Refactor the frond end part of tags:

- [x]  Add ability to enter tags by comma or enter

- [x] Add ability to rearrange tags (drag and drop)

- [ ] Make the tags fully keyboard accessible

- [ ] Implement both ways of entering tags (through a given list or by typing)

- [ ] New design (on hold)

- [x] Drop all dependencies (chosen)

- [x] Make it a custom element

- [ ] Adjust backend to this changes

### Actual result
So far: tags load form the db and rendered correctly, typed tags are not saved in the db:
screen shot 2018-02-11 at 20 35 16

### Documentation Changes Required
Yes

### CODE REVIEW

**Comment by brianteeman** [comment]
Re the draggable ordering? Is there any reason for adding this functionality iirc the order of tags has no relevance

File: build/webcomponents/js/field-tag/field-tag.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	}
+
+	appendNewTag(text, value) {
+		const self = this;
+		const para = document.createElement('span');
+		para.className = 'tag delete';
+		para.setAttribute('tabindex', 0);
+		para.draggable = true;
+		para.addEventListener('keyup', (e) => {
+			if (e.keyCode === 8 || e.keyCode === 46) {
```
Comment (brianteeman) [inline build/webcomponents/js/field-tag/field-tag.js:208]: maybe i am wrong but i thought keycode was deprecated and no longer recommended and you are supposed to use keypress instead

**Comment by dgrammatiko** [comment]
@brianteeman personally I'm not ok knowing that I cannot rearrange some data once saved. Probably low priority but I think is needed if we want to say that this component is feature complete

**Comment by brianteeman** [comment]
the question is "does rearranging the order of the tags have any effect"

**Comment by mbabker** [comment]
You'd just need to save the order mapping.  And for those who would prefer to be able to order these things instead of relying on input or creation order, it's a welcome minor feature.

**Comment by brianteeman** [comment]
i bring it up because we have had issues raised in the past that the content is not sorted by the ordering of the tags

**Comment by mbabker** [comment]
That's still out of scope for this PR.  At most this PR would introduce support for setting the display order on a content item's tags.  Sorting content by tag is a much more complex task.

**Comment by brianteeman** [comment]
Thats exactly why I asked the purpose

**Comment by dgrammatiko** [comment]
@brianteeman one very dumb example how that small change could be useful:
in the layout you check the tags and if some specific tag is first you do something differently

But as @mbabker this is solely the order that tags are presented in one item/article/etc, nothing more

**Comment by brianteeman** [comment]
you've missed what i am saying. oh well, guess i need a better way of communicating

File: build/webcomponents/js/field-tag/field-tag.js (hunk 1 above)
Comment (C-Lodder) [reply build/webcomponents/js/field-tag/field-tag.js:208]: Yup, this should be: 

`if (e.key === 'Backspace' || e.key === 'Delete')`

Comment (brianteeman) [reply build/webcomponents/js/field-tag/field-tag.js:208]: woot I got something (almost) right on this PR

Comment (dgrammatiko) [reply build/webcomponents/js/field-tag/field-tag.js:208]: MS:
d4042f1b-2b2b-4830-aa8d-c87713600595

Comment (brianteeman) [reply build/webcomponents/js/field-tag/field-tag.js:208]: Read the note - the MS issue is only for arrow keys - so no issue here ;)

**Comment by wilsonge** [comment]
I'm closing this for now given it's WIP that hasn't been worked on since February, there's hound errors, conflicts etc and for now our agreement to focus on the existing elements before we introduce new ones last night. Definitely we can come back to this if there is time.

**Pull request closed without merging**